test/ts/unit/debt_token.ts

```diff
@@ -1429,4 +1437,125 @@ contract("Debt Token (Unit Tests)", (ACCOUNTS) => {
+                { from: from },
+            );
+        };
+
+        const shouldSafelyTransfer = async (to?: string) => {
+            describe("with data", () => {
+                it("should call `onERC721Received`", async function() {
+                    const data = "0x42";
+                    const tokenID = debtEntries[0].getTokenId();
+                    await safelyTransferWithData(
+                        TOKEN_OWNER_1,
+                        to ? to : receiver.address,
+                        tokenID,
+                        data,
+                    );
+                    expect(
+                        receiver.wasOnERC721ReceivedCalledWith.callAsync(
+                            TOKEN_OWNER_1,
+                            tokenID,
+                            data,
+                        ),
+                    ).to.eventually.be.true;
+                });
+            });
+            describe("without data", () => {
+                it("should call `onERC721Received`", async function() {
+                    const tokenID = debtEntries[0].getTokenId();
+                    await safelyTransferWithoutData(
+                        TOKEN_OWNER_1,
+                        to ? to : receiver.address,
+                        tokenID,
+                    );
+                    expect(
+                        receiver.wasOnERC721ReceivedCalledWith.callAsync(
+                            TOKEN_OWNER_1,
+                            tokenID,
+                            "",
+                        ),
+                    ).to.eventually.be.true;
+                });
+            });
+        };
+
+        const shouldNotSafelyTransfer = async (toReceiver: boolean = true) => {
+            describe("with data", () => {
+                it("should revert", async function() {
+                    const data = "0x42";
+                    const tokenID = debtEntries[0].getTokenId();
+                    expect(
+                        safelyTransferWithData(
+                            TOKEN_OWNER_1,
+                            toReceiver ? receiver.address : mockRegistry.address,
+                            tokenID,
+                            data,
+                        ),
+                    ).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+            });
+            describe("without data", () => {
+                it("should revert", async function() {
+                    const tokenID = debtEntries[0].getTokenId();
+                    expect(
+                        safelyTransferWithoutData(
+                            TOKEN_OWNER_1,
+                            toReceiver ? receiver.address : mockRegistry.address,
+                            tokenID,
+                        ),
+                    ).to.eventually.be.rejectedWith(REVERT_ERROR);
+                });
+            });
+        };
+
+        describe("to a valid receiver contract", () => {
```

Comment:
Please include tests for:

1. Transferring to non-contract accounts
2. Transferring under approval conditions -- i.e. ACCOUNT_1 approved ACCOUNT_2 for X tokens, ACCOUNT_2 calls safeTransferFrom(ACCOUNT_1, Y);

Reply:
Pivotal story here: https://www.pivotaltracker.com/story/show/156604273